weave/Scripts/Main.cs

```diff
@@ -83,6 +84,27 @@ private void DetectPlayerCollision()
             GameOver();
     }
 
+    private void DetectPlayerOutOfBounds()
+    {
+        var width = (int)GetViewportRect().Size.X;
```

Comment:
Could you cache these? (its ok because we have already made assumptions about resolution not changing)